src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -522,6 +599,81 @@ void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view
     }
 }
 
+template<>
+void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view strVal,
+    const CSVReaderConfig& csvReaderConfig, int64_t value) {
+    if (strVal.empty() || isNull(strVal)) {
+        vector->setNull(rowToAdd, true /* isNull */);
+        return;
+    } else {
+        vector->setNull(rowToAdd, false /* isNull */);
+    }
+
+    function::simpleIntegerCast<int64_t>(
+        strVal.data(), strVal.length(), value, LogicalType{LogicalTypeID::INT64});
+    vector->setValue(rowToAdd, value);
+}
+
+template<>
+void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view strVal,
+    const CSVReaderConfig& csvReaderConfig, int32_t value) {
+    if (strVal.empty() || isNull(strVal)) {
+        vector->setNull(rowToAdd, true /* isNull */);
+        return;
+    } else {
+        vector->setNull(rowToAdd, false /* isNull */);
+    }
+
+    function::simpleIntegerCast<int32_t>(
+        strVal.data(), strVal.length(), value, LogicalType{LogicalTypeID::INT32});
+    vector->setValue(rowToAdd, value);
+}
+
+template<>
+void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view strVal,
+    const CSVReaderConfig& csvReaderConfig, int16_t value) {
```

Comment:
The casting for numeric types are very similar: You can implement the casting for numeric types in the template function body, and use specialization for other types.  So you don't need to duplicate the code for numeric type casting.
The numeric type casting:
```
 if (strVal.empty() || isNull(strVal)) {
        vector->setNull(rowToAdd, true /* isNull */);
        return;
    } else {
        vector->setNull(rowToAdd, false /* isNull */);
    }

    function::simpleIntegerCast<T>(
        strVal.data(), strVal.length(), value, vector->getDataType());
    vector->setValue(rowToAdd, value);
```
Using this generic  templated numeric type casting function, you don't need to duplicate the common code for all numeric types.
Anyways, feel free to ping me if you have questions regarding to the refactor. Or you can leave it to me.